backend/src/cusco/sources/pt2030.py

```
@@ -0,0 +1,225 @@
+        CACHE_DIR.mkdir(parents=True, exist_ok=True)
+
+    async def _ensure_loaded(self) -> None:
+        """Idempotent load — serialized via `_load_once` so concurrent
+        first queries don't each re-download + re-parse the dataset."""
+        await self._load_once(self._do_load, lambda: self._loaded)
+
+    async def _do_load(self) -> None:
+        try:
+            rows = await self._load_rows()
+            self._index(rows)
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)